File: vn/src/main/java/org/sqlbuilder/vn/VnModule.java
package org.sqlbuilder.vn;

import org.sqlbuilder.common.Module;
import org.sqlbuilder.vn.collector.VnExportingProcessor;
import org.sqlbuilder.vn.collector.VnProcessor;
import org.sqlbuilder.vn.collector.VnUpdatingProcessor;

import java.io.IOException;

public class VnModule extends Module
{
	public static final String MODULE_ID = "vn";

	public VnModule(final String conf, final Mode mode)
	{
		super(MODULE_ID, conf, mode);
	}

	@Override
	protected void run()
	{
		assert props != null;

		switch(mode)
		{
			case PLAIN:
			case RESOLVE:
				new VnProcessor(props).run();
				try
				{
					Inserter inserter = mode == Mode.PLAIN ? new Inserter(props) : new ResolvingInserter(props);
					inserter.insert();
				}
				catch (IOException | ClassNotFoundException e)
				{
					e.printStackTrace();
				}
				break;

			case UPDATE:
				new VnUpdatingProcessor(props).run();
				try
				{
					Inserter inserter = new ResolvingUpdater(props);
					inserter.insert();
				}
				catch (IOException | ClassNotFoundException e)
				{
					e.printStackTrace();
				}
				break;

			case EXPORT:
				new VnExportingProcessor(props).run();
				try
				{
					Exporter exporter = new Exporter(props);
					exporter.run();
				}
				catch (IOException e)
				{
					e.printStackTrace();
				}
				break;

			default:
				return;
		}
	}

	public static void main(final String[] args) throws IOException
	{
		int i = 0;
		Mode mode = Mode.PLAIN;
		if (args[i].startsWith("-"))
		{
			mode = Mode.read(args[i++]);
		}
		String conf = args[i];
		new VnModule(conf, mode).run();
	}
}

File: vn/src/main/java/org/sqlbuilder/vn/VnModule.kt
package org.sqlbuilder.vn

import org.sqlbuilder.common.Module
import org.sqlbuilder.common.Module.Mode.Companion.read
import org.sqlbuilder.vn.collector.VnCollector
import org.sqlbuilder.vn.collector.VnExportCollector
import org.sqlbuilder.vn.collector.VnUpdateCollector
import java.io.IOException

class VnModule(conf: String, mode: Mode) : Module(MODULE_ID, conf, mode) {

    override fun run() {
        checkNotNull(props)

        when (mode) {
            Mode.PLAIN, Mode.RESOLVE -> {
                VnCollector(props).run()
                try {
                    val inserter = if (mode == Mode.PLAIN) Inserter(props) else ResolvingInserter(props)
                    inserter.insert()
                } catch (e: IOException) {
                    e.printStackTrace()
                } catch (e: ClassNotFoundException) {
                    e.printStackTrace()
                }
            }

            Mode.UPDATE              -> {
                VnUpdateCollector(props).run()
                try {
                    val inserter: Inserter = ResolvingUpdater(props)
                    inserter.insert()
                } catch (e: IOException) {
                    e.printStackTrace()
                } catch (e: ClassNotFoundException) {
                    e.printStackTrace()
                }
            }

            Mode.EXPORT              -> {
                VnExportCollector(props).run()
                try {
                    val exporter = Exporter(props)
                    exporter.run()
                } catch (e: IOException) {
                    e.printStackTrace()
                }
            }

            else                     -> {}
        }
    }

    companion object {

        const val MODULE_ID: String = "vn"

        fun main(args: Array<String>) {
            var i = 0
            var mode = Mode.PLAIN
            if (args[i].startsWith("-")) {
                mode = read(args[i++])
            }
            val conf = args[i]
            VnModule(conf, mode).run()
        }
    }
}
